pkg/kubectl/apply/strategy/replace_visitor.go

```diff
@@ -98,3 +109,42 @@ func (v replaceStrategy) doAddOrDelete(e apply.Element) (apply.Result, bool) {
 }
 
 var _ apply.Strategy = &replaceStrategy{}
+
+func (v replaceStrategy) detectConflict(e apply.Element) (apply.Result, error) {
+	switch e.(type) {
+	case apply.ListElement:
```

Comment:
Why isn't this a function on the specific element type?

Reply:
I originally wanted to add this small function in strategy without impact Element interface and impl. I got your point, so I want to do as following:
1. Add an interface `DetectConflict(Strategy)` in apply/Element.go
2. Implement this interface for each particular element.
Does it make sense ?

Reply:
The simplest thing may be to add the interface for conflict detection and then check if e implements the interface - then call the function.